include/exadg/grid/grid.cpp

```diff
@@ -160,10 +160,47 @@ Grid<dim>::create_triangulation(
     {
       // we need to generate the coarse triangulations explicitly
 
-      // TODO
-      // construct all the coarser triangulations
+      // resize the empty coarse triangulations vector
+      coarse_triangulations = std::vector<std::shared_ptr<dealii::Triangulation<dim> const>>(
+        triangulation->n_global_levels());
+
+      unsigned int coarse_level = triangulation->n_global_levels() - 1;
+
+      // lambda function for creating and placing the coarse triangulations
+      auto const lambda_create_coarse_levels = [&](unsigned int              global_refinement,
+                                                   std::vector<unsigned int> refine_local) {
+        auto new_tria = std::make_shared<dealii::parallel::fullydistributed::Triangulation<dim>>(
+          triangulation->get_communicator());
+
+        do_create_triangulation(
+          new_tria, data, create_coarse_triangulation, global_refinement, refine_local);
+
+        coarse_triangulations[coarse_level] = new_tria;
+
+        coarse_level--;
+      };
 
-      AssertThrow(false, ExcNotImplemented());
+      // start with the finest level
+      std::vector<unsigned int> refine_local = vector_local_refinements;
+
+      // undo global refinements
+      for(int it = global_refinements; it >= 0; --it)
+      {
+        lambda_create_coarse_levels(it, refine_local);
+      }
```

Comment:
```suggestion
      std::vector<unsigned int> refine_local = vector_local_refinements;
      for(auto & i : refine_local)
        i += global_refinements;
```

Reply:
I don't see how this would do the same thing as the original code.

Reply:
Yes that makes sense if we always have a local refinement, but what happens if the `refine_local` vector is empty? Then the `global_refinements` are also disregarded, and no `coarse_triangulations` are created.

Reply:
Can that be empty? If yes let's keep it like this!

Reply:
Yes, it is the default case.

https://github.com/exadg/exadg/blob/7edf07c5a30582accdd72ce2af833caa51382107/include/exadg/grid/grid.h#L68-L73